File: sorting/BubbleSort.java
class BubbleSort{
    static int[] bubbleSort(int[] A){
        for(int i=1;i<A.length;i++){
            for(int j=i;j<A.length;j++){
                if(A[j-1] > A[j]){
                    int temp=A[j-1];
                    A[j-1]=A[j];
                    A[j]=temp;
                }
            }
        }
        return A;
    } 
    public static void main(String[] args){
        int[] A={
            1, 4, 2, 6, 5, 3
        };
        A=bubbleSort(A);
        for(int i=0;i<A.length;i++){
            System.out.print(A[i]+"\t");
        }
    }
}

File: sorting/BubbleSort.cpp
#include <iostream>

using namespace std;

int  bubbleSort(int *A, int n){
    int counter=0;
    for(int i=0;i<n;i++){
        for(int j=0;j<n-i-1;j++){
            if(*(A+j)>*(A+j+1)){
                int temp=*(A+j+1);
                *(A+j+1)=*(A+j);
                *(A+j)=temp;
                counter+=1;
            }
        }
    }
    return counter;
}

int main(){
    int n;
    cin>>n;
    int Arr[n];
    for(int i=0;i<n;i++){
        cin>>Arr[i];
    }
    int *A=Arr;
    int counter = bubbleSort(A, n);
    cout<<counter;
    // for(int i=0;i<n;i++){
    //     cout<<A[i]<<"\t";
    // }
    return 0;
}